File: app/src/main/java/com/example/dictionaryapp/WordDetailActivity.java
package com.example.dictionaryapp;

import android.content.Intent;
import android.os.Build;
import android.os.Bundle;
import android.view.View;
import android.widget.ImageButton;
import android.widget.TextView;

import androidx.annotation.RequiresApi;
import androidx.appcompat.app.AlertDialog;
import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.widget.Toolbar;
import androidx.cardview.widget.CardView;

import com.google.android.material.bottomnavigation.BottomNavigationView;
import com.google.android.material.chip.Chip;
import com.google.android.material.chip.ChipGroup;

import org.json.JSONArray;
import org.json.JSONException;

public class WordDetailActivity extends AppCompatActivity {

    private JSONArray jsonTags;

    @RequiresApi(api = Build.VERSION_CODES.O)
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_word_detail);
        Toolbar toolbar = findViewById(R.id.toolbar);
        setSupportActionBar(toolbar);
        getSupportActionBar().setDisplayHomeAsUpEnabled(true);
        getSupportActionBar().setDisplayShowHomeEnabled(true);

        toolbar.setNavigationOnClickListener(v -> onBackPressed());

        Intent intent = getIntent();
        String kanji = intent.getStringExtra("word");
        String reading = intent.getStringExtra("reading");
        String meaning = intent.getStringExtra("meaning");
        String freq = intent.getStringExtra("freq");
        String pitch = intent.getStringExtra("pitch");
        String tags = intent.getStringExtra("tags");
        jsonTags = new JSONArray();
        try {
            jsonTags = new JSONArray(tags);
        } catch (JSONException e) {
            e.printStackTrace();
        }

        TextView wordTextView = findViewById(R.id.wordTextView);
        TextView meaningTextView = findViewById(R.id.meaningTextView);
        TextView readingTextView = findViewById(R.id.secondaryTextCard);
        TextView pitchTextView = findViewById(R.id.pitchText);
        TextView freqChip = findViewById(R.id.freqChip);

        CardView pitchCard = findViewById(R.id.pitchCard);
        ChipGroup chipView = findViewById(R.id.chipLayout);
        for (int x = 0; x < jsonTags.length(); x++) {
            try {
                JSONArray current = jsonTags.getJSONArray(x);
                Chip chip = new Chip(WordDetailActivity.this);
                chip.setText(current.getString(0));
                chip.setChipStartPadding(30);
                chip.setOnClickListener(v -> {
                    Chip c = (Chip) v;
                    for (int y = 0; y < jsonTags.length(); y++) {
                        try {
                            if (jsonTags.getJSONArray(y).getString(0) == c.getText()) {
                                AlertDialog alertDialog = new AlertDialog.Builder(WordDetailActivity.this).create();
                                alertDialog.setMessage(jsonTags.getJSONArray(y).getString(1));
                                alertDialog.setButton(AlertDialog.BUTTON_NEUTRAL, "OK",
                                        (dialog, which) -> dialog.dismiss());
                                alertDialog.show();
                            }
                        } catch (JSONException e) {
                            e.printStackTrace();
                        }
                    }
                });
                chipView.addView(chip);
            } catch (JSONException e) {
                e.printStackTrace();
            }

        }

        toolbar.setTitle(kanji);
        readingTextView.setText(reading);
        wordTextView.setText(kanji);
        meaningTextView.setText(meaning);
        pitchTextView.setText(pitch);
        freqChip.setText(freq);

        if (pitch == null) {
            pitchCard.setVisibility(View.GONE);
        }

        assert freq != null;
        if (freq.isEmpty()) {
            freqChip.setVisibility(View.GONE);
        }

        if (jsonTags.length() == 0) {
            chipView.setVisibility(View.GONE);
        }

        BottomNavigationView btmNavView = findViewById(R.id.bottom_navigation);
        btmNavView.setSelectedItemId(R.id.search_page);

        ImageButton closeBtn = findViewById(R.id.closeBtn);
        closeBtn.setOnClickListener(v -> finishAffinity());
        btmNavView.setOnNavigationItemSelectedListener(menuItem -> {
            switch (menuItem.getItemId()) {
//                    case R.id.word_page:
//                        Intent intent_word = new Intent(WordDetailActivity.this, WordsActivity.class);
//                        startActivity(intent_word);
//                        return true;
                case R.id.setting_page:
                    Intent intent_settings = new Intent(WordDetailActivity.this, SettingsActivity.class);
                    startActivity(intent_settings);
                    return true;
                case R.id.search_page:
                    Intent intent_search = new Intent(WordDetailActivity.this, MainActivity.class);
                    startActivity(intent_search);
                    return true;
            }
            return true;
        });
    }



}

File: app/src/main/java/com/kamui/rin/WordDetailActivity.kt
package com.kamui.rin

import android.content.DialogInterface
import android.os.Build
import android.os.Bundle
import android.view.View
import android.widget.ImageButton
import android.widget.TextView
import androidx.annotation.RequiresApi
import androidx.appcompat.app.AlertDialog
import androidx.appcompat.app.AppCompatActivity
import androidx.appcompat.widget.Toolbar
import androidx.cardview.widget.CardView
import com.google.android.material.bottomnavigation.BottomNavigationView
import com.google.android.material.chip.Chip
import com.google.android.material.chip.ChipGroup
import com.kamui.rin.util.Tag

class WordDetailActivity : AppCompatActivity() {
    @RequiresApi(api = Build.VERSION_CODES.O)
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_word_detail)
        val toolbar = findViewById<Toolbar>(R.id.toolbar)
        setSupportActionBar(toolbar)
        supportActionBar!!.setDisplayHomeAsUpEnabled(true)
        supportActionBar!!.setDisplayShowHomeEnabled(true)
        toolbar.setNavigationOnClickListener { onBackPressed() }

        val kanji = intent.getStringExtra("word")
        val reading = intent.getStringExtra("reading")
        val meaning = intent.getStringExtra("meaning")
        val freq = intent.getStringExtra("freq")
        val pitch = intent.getStringExtra("pitch")
        val tags = intent.getParcelableArrayListExtra<Tag>("tags")!!

        val wordTextView = findViewById<TextView>(R.id.wordTextView)
        val meaningTextView = findViewById<TextView>(R.id.meaningTextView)
        val readingTextView = findViewById<TextView>(R.id.secondaryTextCard)
        val pitchTextView = findViewById<TextView>(R.id.pitchText)
        val freqChip = findViewById<TextView>(R.id.freqChip)
        val pitchCard = findViewById<CardView>(R.id.pitchCard)
        val chipView = findViewById<ChipGroup>(R.id.chipLayout)

        for (tag in tags) {
            val chip = Chip(this@WordDetailActivity)
            chip.text = tag.name
            chip.chipStartPadding = 30f
            chip.setOnClickListener {
                val alertDialog = AlertDialog.Builder(this@WordDetailActivity).create()
                alertDialog.setMessage(tag.description)
                alertDialog.setButton(
                    AlertDialog.BUTTON_NEUTRAL, "OK"
                ) { dialog: DialogInterface, _: Int -> dialog.dismiss() }
                alertDialog.show()
            }
            chipView.addView(chip)
        }

        toolbar.title = kanji
        readingTextView.text = reading
        wordTextView.text = kanji
        meaningTextView.text = meaning
        pitchTextView.text = pitch
        freqChip.text = freq
        if (pitch == null) {
            pitchCard.visibility = View.GONE
        }
        if (freq == null) {
            freqChip.visibility = View.GONE
        }
        if (tags.isEmpty()) {
            chipView.visibility = View.GONE
        }

        val closeBtn = findViewById<ImageButton>(R.id.closeBtn)
        closeBtn.setOnClickListener { finishAffinity() }
    }
}